### ISSUE TITLE
Running MACS2 on RNA-Seq BAM files

### ISSUE THREAD

**Issue by gireeshkbogu** [taoliu/MACS#186]
Hi,

I am aware that MACS2 is designed for finding peaks using ChIP-Seq data. However, when I run MACS2 on RNA-Seq data using default parameters, it seems to works as I expected. Now my question is do you see any problem with this approach? 

Thanks

**Comment by IanCodes** [comment]
Just bear in mind that you are only analysing reads that map to the
transcriptome.  I don't think the assumptions made in the lambda background
model hold true as you are only considering a non contiguous subset of the
genome.  I suppose the other question is what do the results mean?  There
is also no control.  I think my rambling reply boils down to the
question... why use MACS2 on RNA-seq data?  :)

**Comment by gireeshkbogu** [comment]
@IanCodes Thank you for the quick comment. What do you mean non contiguous set? and do you mean it is better to ignore lambda modelling while finding peaks using RNA-Seq?

#Why use MACS2 on RNA-Seq 
I am trying to find genomic repeats with reliable expression patterns. Initially, I did this just by counting overlapping RNA-Seq reads with genomic repeats but with this approach, most of the expressed repeats seems to fall in regions where the RNA-Seq signal is flat . I hypothesised that if a TE is expressed it should have some kind of peak pattern depending upon where it is in the genome. Here is an example that gives a big picture 
![screenshot](https://cloud.githubusercontent.com/assets/3885659/24951044/5f1d8190-1f73-11e7-9ba3-06c093e601db.png). In the highlighted figure the black square is the repeat elements and green color indicates RNA-Seq peak found by MACS2 .

**Comment by IanCodes** [comment]
Sorry I wasn't clear.  MACS2 generates a background model looking within
the enriched region, 1,000K and 10,000 from the region, and across the
whole genome.  However, the transcriptome coverage is patchy across the
genome, so I think the lambda background would be estimated incorrectly.
You can turn off the modeling, using --nolambda (i think).

Now you have explained you aims then MACS2 could give you what you need to
find candidates, a bit like an ATAC-seq analysis that also does not use a
control.  It might be worth screening any output using the ENCODE blacklist
regions, if available for this genome.

**Comment by gireeshkbogu** [comment]
@IanCodes 

ENCODE 'blacklist' means unmappable regions?

Thank you for emphasising the lambda modelling. I will test how this lambda modelling affects the no.of peaks and will decide whether to use it or not.

**Comment by IanCodes** [comment]
It is more complicated than mappability, see:
https://sites.google.com/site/anshulkundaje/projects/blacklists

Although reading the summary suggest they are not applicable to RNA-seq
data.  But then as these are transcriptome associated reads you might be OK.

**Comment by gireeshkbogu** [comment]
Yes, I think this black list won't affect RNA-Seq analysis. Thanks @IanCodes !

**Comment by genecell** [comment]
Hi Gireesh @gireeshkbogu, I am also trying to call peaks for RNA-seq datasets using MACS2, could you please share me with the parameter settings? Really appreciated! Thank you!

Best regards,
Min

**Comment by Kiliankleemann** [comment]
@gireeshkbogu @genecell Same!

**Comment by amritgaire-umn** [comment]
Hi @gireeshkbogu @genecell @Kiliankleemann , I am also trying to call peaks for RNA-seq datasets using MACS2, could you please share me with the parameter settings? Really appreciated! Thank you!

Best,
AG
